packages/cli/api-importers/v2-importer-commons/src/converters/schema/SchemaConverter.ts

```diff
@@ -173,11 +193,209 @@ export class SchemaConverter extends AbstractConverter<AbstractConverterContext<
         };
     }
 
-    public convertDeclaredTypeName({ context }: { context: AbstractConverterContext<object> }): DeclaredTypeName {
+    public convertDeclaredTypeName({
+        context
+    }: {
+        context: AbstractConverterContext<object>;
+    }): FernIr.DeclaredTypeName {
         return {
             typeId: this.id,
             fernFilepath: context.createFernFilepath(),
             name: context.casingsGenerator.generateName(this.id)
         };
     }
+
+    private async createTypeDeclarationFromFernType({
+        fernType,
+        context,
+        errorCollector
+    }: {
+        fernType: string;
+        context: AbstractConverterContext<object>;
+        errorCollector: ErrorCollector;
+    }): Promise<FernIr.TypeDeclaration | undefined> {
+        const name = this.convertDeclaredTypeName({ context });
+        const docs = this.schema.description;
+        const availability = await context.getAvailability({
+            node: this.schema,
+            breadcrumbs: this.breadcrumbs,
+            errorCollector
+        });
+        const referencedTypes = new Set<string>();
+        const inline = this.inlined;
+        const baseTypeDeclaration = {
+            name,
+            autogeneratedExamples: [],
+            userProvidedExamples: [],
+            encoding: undefined,
+            availability,
+            docs,
+            referencedTypes,
+            source: undefined,
+            inline
+        };
```

Comment:
Discussed offline - we can write these properties inline here since they aren't used elsewhere.